logger output missing timestamp values

### Consul Template version

Vault agent version 1.7.0, importing **consul-template version 0.25.2**

### Configuration

N/A

### Command

See debug output of command + logs

### Debug output

```
               Log Level: info
                 Version: Vault v1.4.0

2021-05-26T22:46:07.569-0700 [INFO]  sink.file: creating file sink
2021-05-26T22:46:07.570-0700 [INFO]  sink.file: file sink configured: path=/tmp/agent_test/.vault-token mode=-rw-r-----
2021-05-26T22:46:07.570-0700 [INFO]  sink.server: starting sink server
2021-05-26T22:46:07.570-0700 [INFO]  auth.handler: starting auth handler
2021-05-26T22:46:07.570-0700 [INFO]  template.server: starting template server
2021-05-26T22:46:07.570-0700 [INFO]  auth.handler: authenticating
2021/05/27 05:46:07.571269 [INFO] (runner) creating new runner (dry: false, once: false)
2021/05/27 05:46:07.572112 [INFO] (runner) creating watcher
2021-05-26T22:46:07.683-0700 [INFO]  auth.handler: authentication successful, sending token to sinks
2021-05-26T22:46:07.683-0700 [INFO]  template.server: template server received new token
2021-05-26T22:46:07.683-0700 [INFO]  auth.handler: starting renewal process
2021/05/27 05:46:07.683871 [INFO] (runner) stopping
2021/05/27 05:46:07.683927 [INFO] (runner) creating new runner (dry: false, once: false)
2021/05/27 05:46:07.684056 [INFO] (runner) creating watcher
2021-05-26T22:46:07.684-0700 [INFO]  sink.file: token written: path=/tmp/agent_test/.vault-token
2021/05/27 05:46:07.684242 [INFO] (runner) starting
2021-05-26T22:46:07.688-0700 [INFO]  auth.handler: renewed auth token
2021/05/27 05:46:07.693380 [WARN] (view) vault.read(kv/b): no secret exists at kv/b (retry attempt 1 after "250ms")
2021/05/27 05:46:07.953052 [WARN] (view) vault.read(kv/b): no secret exists at kv/b (retry attempt 2 after "500ms")
2021/05/27 05:46:08.463408 [WARN] (view) vault.read(kv/b): no secret exists at kv/b (retry attempt 3 after "1s")
2021/05/27 05:46:09.470984 [WARN] (view) vault.read(kv/b): no secret exists at kv/b (retry attempt 4 after "2s")
2021/05/27 05:46:11.480051 [WARN] (view) vault.read(kv/b): no secret exists at kv/b (retry attempt 5 after "4s")
2021/05/27 05:46:15.485882 [WARN] (view) vault.read(kv/b): no secret exists at kv/b (retry attempt 6 after "8s")
2021/05/27 05:46:23.496498 [WARN] (view) vault.read(kv/b): no secret exists at kv/b (retry attempt 7 after "16s")
2021/05/27 05:46:39.504152 [WARN] (view) vault.read(kv/b): no secret exists at kv/b (retry attempt 8 after "32s")
2021/05/27 05:47:11.512512 [WARN] (view) vault.read(kv/b): no secret exists at kv/b (retry attempt 9 after "1m0s")
2021/05/27 05:48:11.519200 [WARN] (view) vault.read(kv/b): no secret exists at kv/b (retry attempt 10 after "1m0s")
2021/05/27 05:49:11.524638 [WARN] (view) vault.read(kv/b): no secret exists at kv/b (retry attempt 11 after "1m0s")
2021/05/27 05:50:11.533526 [WARN] (view) vault.read(kv/b): no secret exists at kv/b (retry attempt 12 after "1m0s")
2021/05/27 05:51:11.541864 [ERR] (view) vault.read(kv/b): no secret exists at kv/b (exceeded maximum retries)
2021/05/27 05:51:11.541882 [ERR] (runner) watcher reported error: vault.read(kv/b): no secret exists at kv/b
2021-05-26T22:51:11.541-0700 [ERROR] template.server: template server error: error="vault.read(kv/b): no secret exists at kv/b"
2021-05-26T22:51:11.541-0700 [INFO]  template.server: template server stopped
^C==> Vault agent shutdown triggered
```

1.7.0
```
==> Vault agent configuration:

                     Cgo: disabled
               Log Level: info
                 Version: Vault v1.7.0
             Version Sha: 4e222b85c40a810b74400ee3c54449479e32bb9f

2021-05-26T22:39:49.086-0700 [INFO]  sink.file: creating file sink
2021-05-26T22:39:49.086-0700 [INFO]  sink.file: file sink configured: path=/tmp/agent_test/.vault-token mode=-rw-r-----
2021-05-26T22:39:49.087-0700 [INFO]  sink.server: starting sink server
2021-05-26T22:39:49.087-0700 [INFO]  auth.handler: starting auth handler
2021-05-26T22:39:49.087-0700 [INFO]  auth.handler: authenticating
2021-05-26T22:39:49.087-0700 [INFO]  template.server: starting template server
[INFO] (runner) creating new runner (dry: false, once: false)
[INFO] (runner) creating watcher
2021-05-26T22:39:49.202-0700 [INFO]  auth.handler: authentication successful, sending token to sinks
2021-05-26T22:39:49.202-0700 [INFO]  auth.handler: starting renewal process
2021-05-26T22:39:49.202-0700 [INFO]  template.server: template server received new token
[INFO] (runner) stopping
[INFO] (runner) creating new runner (dry: false, once: false)
[INFO] (runner) creating watcher
[INFO] (runner) starting
2021-05-26T22:39:49.203-0700 [INFO]  sink.file: token written: path=/tmp/agent_test/.vault-token
2021-05-26T22:39:49.204-0700 [INFO]  auth.handler: renewed auth token
[WARN] (view) vault.read(kv/b.v1): no secret exists at kv/b (retry attempt 1 after "250ms")
[WARN] (view) vault.read(kv/b.v1): no secret exists at kv/b (retry attempt 2 after "500ms")
[ERR] (view) vault.read(kv/b.v1): no secret exists at kv/b (exceeded maximum retries)
[ERR] (runner) watcher reported error: vault.read(kv/b.v1): no secret exists at kv/b
[INFO] (runner) stopping
2021-05-26T22:39:49.969-0700 [INFO]  template.server: template server stopped
2021-05-26T22:39:49.969-0700 [INFO]  auth.handler: shutdown triggered, stopping lifetime watcher
2021-05-26T22:39:49.969-0700 [INFO]  sink.server: sink server stopped
2021-05-26T22:39:49.969-0700 [INFO]  sinks finished, exiting
2021-05-26T22:39:49.969-0700 [INFO]  auth.handler: auth handler stopped
2021-05-26T22:39:49.969-0700 [ERROR] runtime error encountered: error="template server: vault.read(kv/b.v1): no secret exists at kv/b"
```

### Expected behavior

Logs by the consul-template library should have timestamp formatted output

### Actual behavior

Timestamps are entirely missing after upgrading to v0.25.2.

I suspect that it may be because 

https://github.com/hashicorp/consul-template/blob/980d053344b565728b8fd884b0d888cef6f7daf6/logging/logging.go#L75

is always overwriting the log output set earlier in

https://github.com/hashicorp/consul-template/blob/980d053344b565728b8fd884b0d888cef6f7daf6/logging/logging.go#L46

### Steps to reproduce

1. Import consul-template library
2. Render secrets and output it logs
3. Observe difference in log output

### References

Are there any other GitHub issues (open or closed) that should
be linked here? For example:
- https://github.com/hashicorp/consul-template/pull/1412
- https://github.com/hashicorp/consul-template/pull/1454

--- a/logging/logging.go
+++ b/logging/logging.go
@@ -12,15 +12,27 @@ import (
 	"github.com/hashicorp/logutils"
 )
 
+// Log time format
+const timeFmt = "2006-01-02T15:04:05.000Z0700"
+
 // Levels are the log levels we respond to=o.
 var Levels = []logutils.LogLevel{"TRACE", "DEBUG", "INFO", "WARN", "ERR"}
 
 type logWriter struct {
+	out io.Writer
+}
+
+// To let me replace in tests
+var now = func() string {
+	return time.Now().Format(timeFmt)
 }
 
 // writer to output date / time in a standard format
 func (writer logWriter) Write(bytes []byte) (int, error) {
-	return fmt.Print(time.Now().Format("2006-01-02T15:04:05.000Z0700") + " " + string(bytes))
+	if len(bytes) == 0 {
+		return 0, nil
+	}
+	return fmt.Fprintf(writer.out, "%s %s", now(), bytes)
 }
 
 // Config is the configuration for this log setup.
@@ -40,55 +52,67 @@ type Config struct {
 }
 
 func Setup(config *Config) error {
-	var logOutput io.Writer
+	logOutput, err := newWriter(config)
+	if err != nil {
+		return err
+	}
 
 	log.SetFlags(0)
-	log.SetOutput(new(logWriter))
-
-	// Setup the default logging
-	logFilter := NewLogFilter()
-	logFilter.MinLevel = logutils.LogLevel(strings.ToUpper(config.Level))
-	logFilter.Writer = config.Writer
-	if !ValidateLevelFilter(logFilter.MinLevel, logFilter) {
-		levels := make([]string, 0, len(logFilter.Levels))
-		for _, level := range logFilter.Levels {
-			levels = append(levels, string(level))
-		}
-		return fmt.Errorf("invalid log level %q, valid log levels are %s",
-			config.Level, strings.Join(levels, ", "))
-	}
-
-	// Check if syslog is enabled
-	if config.Syslog {
-		log.Printf("[DEBUG] (logging) enabling syslog on %s", config.SyslogFacility)
-
-		l, err := gsyslog.NewLogger(gsyslog.LOG_NOTICE, config.SyslogFacility, config.SyslogName)
-		if err != nil {
-			return fmt.Errorf("error setting up syslog logger: %s", err)
-		}
-		syslog := &SyslogWrapper{l, logFilter}
-		logOutput = io.MultiWriter(logFilter, syslog)
-	} else {
-		logOutput = io.MultiWriter(logFilter)
-	}
-
 	log.SetOutput(logOutput)
 
 	return nil
 }
 
-// NewLogFilter returns a LevelFilter that is configured with the log levels that
-// we use.
-func NewLogFilter() *logutils.LevelFilter {
-	return &logutils.LevelFilter{
-		Levels:   Levels,
-		MinLevel: "WARN",
-		Writer:   ioutil.Discard,
+// Creates a log writer w/ filtering
+func newWriter(config *Config) (io.Writer, error) {
+	var logOutput io.Writer = logWriter{out: config.Writer}
+	logLevel := logutils.LogLevel(strings.ToUpper(config.Level))
+
+	logOutput, err := newLogFilter(logOutput, logLevel)
+	if err != nil {
+		return nil, err
 	}
+
+	if config.Syslog {
+		log.Printf("[DEBUG] (logging) enabling syslog on %s", config.SyslogFacility)
+
+		l, err := gsyslog.NewLogger(gsyslog.LOG_NOTICE, config.SyslogFacility, config.SyslogName)
+		if err != nil {
+			return nil, fmt.Errorf("error setting up syslog logger: %s", err)
+		}
+		syslog := &SyslogWrapper{l, logOutput.(*logutils.LevelFilter)}
+		logOutput = io.MultiWriter(logOutput, syslog)
+	}
+
+	return logOutput, nil
 }
 
-// ValidateLevelFilter verifies that the log levels within the filter are valid.
-func ValidateLevelFilter(min logutils.LogLevel, filter *logutils.LevelFilter) bool {
+// NewLogFilter returns a LevelFilter that is configured with the log levels that
+// we use.
+func newLogFilter(out io.Writer, logLevel logutils.LogLevel) (*logutils.LevelFilter, error) {
+	if out == nil {
+		out = ioutil.Discard
+	}
+
+	logFilter := &logutils.LevelFilter{
+		Levels:   Levels,
+		MinLevel: logLevel,
+		Writer:   out,
+	}
+
+	if !validateLevelFilter(logLevel, logFilter) {
+		levels := make([]string, 0, len(logFilter.Levels))
+		for _, level := range logFilter.Levels {
+			levels = append(levels, string(level))
+		}
+		return nil, fmt.Errorf("invalid log level %q, valid log levels are %s",
+			logLevel, strings.Join(levels, ", "))
+	}
+	return logFilter, nil
+}
+
+// validateLevelFilter verifies that the log levels within the filter are valid.
+func validateLevelFilter(min logutils.LogLevel, filter *logutils.LevelFilter) bool {
 	for _, level := range filter.Levels {
 		if level == min {
 			return true

--- a/logging/logging_test.go
+++ b/logging/logging_test.go
@@ -1,0 +1,106 @@
+package logging
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"strings"
+	"testing"
+	"time"
+)
+
+func TestNow(t *testing.T) {
+	checkTime := func(timeStamp string) {
+		logTime, err := time.Parse(timeFmt, timeStamp)
+		if err != nil {
+			t.Fatal("log time failed to parse:", err)
+		}
+		if time.Now().Before(logTime) {
+			t.Fatal("log happened in the future?")
+		}
+	}
+	// we just want to be sure now() returns a valid date string
+	checkTime(now())
+
+	// pull apart log message and check timestamp
+	var buf bytes.Buffer
+	config := newConfig(&buf)
+	writer, err := newWriter(config)
+	if err != nil {
+		t.Fatal(err)
+	}
+	writer.Write([]byte("XXX"))
+	logSplit := strings.Split(buf.String(), " ")
+	checkTime(logSplit[0])
+	if logSplit[1] != "XXX" {
+		t.Fatalf("Where'd the XXX go?\n(%#v)", logSplit)
+	}
+}
+
+func newConfig(w io.Writer) *Config {
+	return &Config{
+		Level:  "INFO",
+		Writer: w,
+	}
+}
+
+func TestWriter(t *testing.T) {
+	// mock/de-mock now() func
+	defer func(orig func() string) { now = orig }(now)
+	now = func() string { return "*NOW*" }
+
+	type testCase struct {
+		name          string
+		input, output string
+	}
+	runTest := func(tc testCase) {
+		var buf bytes.Buffer
+		config := newConfig(&buf)
+		writer, err := newWriter(config)
+		if err != nil {
+			t.Fatal(err)
+		}
+		writer.Write([]byte(tc.input))
+		if buf.String() != tc.output {
+			t.Fatalf("unexpected log output string: '%s'", buf.String())
+		}
+	}
+
+	for i, tc := range []testCase{
+		{
+			name:   "null",
+			input:  "",
+			output: "",
+		},
+		{
+			name:   "err",
+			input:  "[ERR] (test) should write",
+			output: "*NOW* [ERR] (test) should write",
+		},
+		{
+			name:   "warn",
+			input:  "[WARN] (test) should write",
+			output: "*NOW* [WARN] (test) should write",
+		},
+		{
+			name:   "info",
+			input:  "[INFO] (test) should write",
+			output: "*NOW* [INFO] (test) should write",
+		},
+		{
+			name:   "debug",
+			input:  "[DEBUG] (test) should not write",
+			output: "",
+		},
+		{
+			name:   "trace",
+			input:  "[TRACE] (test) should not write",
+			output: "",
+		},
+	} {
+		t.Run(fmt.Sprintf("%d_%s", i, tc.name),
+			func(t *testing.T) {
+				runTest(tc)
+			})
+	}
+}

--- a/logging/syslog_test.go
+++ b/logging/syslog_test.go
@@ -1,6 +1,7 @@
 package logging
 
 import (
+	"io/ioutil"
 	"os"
 	"runtime"
 	"testing"
@@ -26,8 +27,10 @@ func TestSyslogFilter(t *testing.T) {
 		t.Fatalf("err: %s", err)
 	}
 
-	filt := NewLogFilter()
-	filt.MinLevel = logutils.LogLevel("INFO")
+	filt, err := newLogFilter(ioutil.Discard, logutils.LogLevel("INFO"))
+	if err != nil {
+		t.Fatal(err)
+	}
 
 	s := &SyslogWrapper{l, filt}
 	n, err := s.Write([]byte("[INFO] test"))

